metaseq/data/filter_dataset.py

```diff
@@ -44,13 +44,16 @@ def __init__(self, dataset, frac_data, metric_data, dataset_name_to_index):
         self.concat_dataset = dataset
         self.metric_data = metric_data
 
+
         # We only ever include stuff that is in metric_data. If our actual training set is a superset - we don't care.
         limit = int(np.ceil(len(self.metric_data) * self.frac_data))
-        self.length = limit
 
         self.metric_data.sort_values('metric', inplace=True, ascending=False)
         self.metric_data = self.metric_data[:limit]
 
+        # If there are a subset of data points in the csv file, then just train on those data points
+        # otherwise, take the limit defined by `frac_data`
+        self.length = len(self.metric_data)
```

Comment:
the length of the dataset should be whatever we have in the metric_data, not the theoretical limit (theoretical limit = frac_data * len(previous dataset)